app/liquid/liquid_tag_finder.rb

```diff
@@ -47,6 +66,14 @@ def strip_quotes(str)
     str.gsub(/\A["']|["']\Z/, '')
   end
 
+  # regex: a regex with a capturing group. returns the captured text in first string to match the regex
+  # candidates: an array of strings to search for the regex in
+  def string_search(regex, candidates)
+    matches = candidates.map{ |comment| comment.match(regex) }
+    re_sult = matches.select{ |m| m.present? && m[1].length }.first
+    re_sult.present? ? re_sult[1].strip : nil
```

Comment:
heheh I didn't want to argue, I just disagree stylistically - I find the imperative style with intermediate variable easier to read and reason about. of course, that's why I'm basically making the same kind of [stylistic suggestions in the other direction](https://github.com/SumOfUs/Champaign/pull/304#discussion_r48858484) which I would say you're welcome to ignore too